dds/DCPS/XTypes/TypeLookupService.h

```diff
@@ -73,6 +85,22 @@ class OpenDDS_Dcps_Export TypeLookupService : public virtual DCPS::RcObject {
   void cache_type_info(const DDS::BuiltinTopicKey_t& key, const TypeInformation& type_info);
   const TypeInformation& get_type_info(const DDS::BuiltinTopicKey_t& key) const;
 
+  static bool parse_extensibility(const DCPS::String& x,
+                                  DDS::ExtensibilityKind& extensibility);
+  static const char* extensibility_to_string(DDS::ExtensibilityKind k);
+
+  void default_enum_extensibility(DDS::ExtensibilityKind ek)
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+    default_enum_extensibility_ = ek;
+  }
+
+  void default_bitmask_extensibility(DDS::ExtensibilityKind ek)
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+    default_bitmask_extensibility_ = ek;
+  }
```

Comment:
The default enum and bitmask extensibility in opendds_idl seems to be fixed to `APPENDABLE` but is configurable here. What is an example use case for this?

Reply:
As a user, you may interoperate with a system that is not setting the extensibility correctly, e.g., a new and old OpenDDS.  In this case, it allows the newer system to set the flags.